File: mns-core/src/main/java/cz/fb/manaus/core/manager/filter/MatchedAmountFilter.java
package cz.fb.manaus.core.manager.filter;

import cz.fb.manaus.core.model.Market;
import cz.fb.manaus.core.model.Price;
import org.springframework.stereotype.Component;

import java.util.Set;

@Component
public class MatchedAmountFilter implements MarketFilter {

    @Override
    public boolean accept(Market market, Set<String> categoryBlacklist) {
        var matchedAmount = market.getMatchedAmount();
        return matchedAmount == null || !Price.amountEq(matchedAmount, 0d);
    }

}

File: mns-core/src/main/java/cz/fb/manaus/core/manager/filter/MatchedAmountFilter.kt
package cz.fb.manaus.core.manager.filter

import cz.fb.manaus.core.model.Market
import cz.fb.manaus.core.model.amountEq
import cz.fb.manaus.core.provider.ProviderTag
import org.springframework.stereotype.Component

@Component
class MatchedAmountFilter : FreshMarketFilter {

    override val tags get() = setOf(ProviderTag.MatchedAmount)

    override fun accept(market: Market): Boolean {
        val matchedAmount = market.matchedAmount
        return !(matchedAmount amountEq 0.0)
    }
}